modules/drivers/mongo/src/metabase/driver/mongo/query_processor.clj

```diff
@@ -1180,6 +1197,22 @@
     (for [i (range (apply max (map count posts)))]
       (into {} (map #(get % i)) posts))))
 
+(defn- generate-window-output-clause
+  "Takes a pair of [output-name input-name] and generates an output clause suitable for
+  including in a `$setWindowFields` output block."
+  [[output-name input-name]]
+  {output-name
+   {$sum input-name
+    "window" {"documents" ["unbounded" "current"]}}})
+
+(defn- generate-window-accumulators
+  "Takes a map of {output-name input-name ...} and generates a `$setWindowFields` stage that
+  produces a cumulative sum of those fields."
+  [window-vals]
+  [{$setWindowFields
+    {"sortBy" {"_id" 1}
```

Comment:
This is more complicated, I'm afraid. Sometimes there's no need to sort at all, sometimes partitioning is needed. Luckily, there are failing tests that help to point out these things.